invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/ResultsTableBody.js

```diff
@@ -20,21 +13,21 @@ export default class ResultsTableBody extends Component {
           onClick={() => {
             this.props.rowActionClickHandler(row);
           }}
-          data-test={'btn-view-details-' + row.ID}
+          data-test={'btn-view-details-' + row.metadata.pid}
         />
       ) : null;
 
       return (
-        <Table.Row key={row.ID} data-test={row.ID}>
+        <Table.Row key={row.metadata.pid} data-test={row.metadata.pid}>
           <Table.Cell textAlign="center">{withRowAction}</Table.Cell>
-          {columns.map((column, idx) =>
-            this.renderCell(
-              row[column] ? row[column] : '-',
-              column,
-              row.ID,
-              idx
-            )
-          )}
+          {columns.map(col => (
+            <Table.Cell
+              key={`${col.title}-${row.metadata.pid}`}
+              data-test={`${col.title}-${row.metadata.pid}`}
+            >
+              {_get(row, col.field)}
```

Comment:
In case we have your suggestion about the `formatter` option I would call it with `formatter(_get(row, col.field)` so the formatter I would expect to be binded to the value you specify from the `field` and return a representation of this value. But in general I think is a nice simple solution to format each field specifically in case you need :)

Reply:
@zzacharo I am not sure I understand what you mean here. 
If it was about the callback function, I think @topless did not implement yet.
About the function itself, if I understand correctly, you would like to have only that specific field and not the entire row in the parameter? If yes, I think it might be better to keep the entire row, in case you need to format the value based on other fields.
Probably the signature would be something like (I would add the index too):
```
render(index, value, [row])
```
The index is nice if you want to, for example, highlight odd/even rows or the first/last.
WDYT?

Reply:
@ntarocco yes I was trying to think if you need a case like the one you described but it didn't come to my mind something. I am totally fine with that.

Reply:
@zzacharo passing only the specific property is what I understood but the point is as @ntarocco mentioned that we want access to the whole object and not only the ID as is the case now. The index might be handy too.